pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -404,6 +405,23 @@ public class ModelInterpreter implements Serializable {
     }
 
     /**
+     *
+     */
+    def evaluateWhen(StageConditionals when) {
+        if (when == null) {
+            return true
+        } else {
+            for (int i = 0; i < when.conditions.size(); i++) {
+                DeclarativeStageConditional c = when.conditions.get(i)
+                if (!c.getScript(script).evaluate()) {
```

Comment:
this seems rather excessive to me, the only benefit I can see is my comment above on the boolean expression.